src/ui/Facet/FacetValues.ts

```diff
@@ -320,8 +215,8 @@ export class FacetValues {
     });
     if (excludeLastIndex != null && excludeLastIndex < this.values.length) {
       var nbExclude = this.getExcluded().length;
-      var excludes = this.values.splice(this.values.length - nbExclude, nbExclude);
-      Array.prototype.splice.apply(this.values, (<any[]>[excludeLastIndex - nbExclude, 0]).concat(excludes));
+      var excludedValues = this.values.splice(this.values.length - nbExclude, nbExclude);
+      Array.prototype.splice.apply(this.values, (<any[]>[excludeLastIndex - nbExclude, 0]).concat(excludedValues));
```

Comment:
I spent some time on this line but ultimately could not understand it. In particular how the following part is interacting with the `splice` method:

```
(<any[]>[excludeLastIndex - nbExclude, 0]).concat(excludedValues)
```

I added unit tests to cover the `sortValuesDependingOnStatus`, but they do not cover the block starting on line 216. If I pass two facet values A, B, where B is excluded, and `excludeLastIndex` is `1`, the returned order is B,A?

Reply:
It looks like
```ts
Array.prototype.splice.apply(this.values, (<any[]>[excludeLastIndex - nbExclude, 0]).concat(excludedValues));
```
is the direct equivalent of
```ts
this.values.splice(
  excludeLastIndex - nbExclude,
  0,
  ...excludedValues
);
```
which appends elements right after the index `excludeLastIndex - nbExclude` without removing any element.

Reply:
I built a small test to confirm this behaviour: https://repl.it/repls/OverdueGrubbyDevelopment

Note that `splice` in this case will return an unchanged array.

Reply:
Thanks Benjamin, that was helpful 💯  I adjusted the code to your simplified example and wrote a test!
What this block is doing is making sure the excluded values are in the viewable part of the list. Since they are sorted at the end, they sometimes need to be 1) spliced out and 2) spliced back into the middle of the array.